### ISSUE TITLE
Allow unlimited groups

### ISSUE THREAD

**Issue by sebastianruizva** [jtackaberry/reaticulate#97]
Currently, we can only use 4 groups in every instance, but several instruments would benefit from more than 4. 

For example, the Friedlander Violin needs 5:

G#1: Sus/stacc/pizz/trem
G#2: legato/poly
G#3: Slur/on/off
G#4: sul pont/sul tasto
G#5 sordino on/off

It would be great if this was possible.

Note: Please let me know if I am misunderstanding the usage of groups.

**Comment by jtackaberry** [comment]
I think I can extend it to 8 groups without too much difficulty.  Maybe even 16 (would have to check the source).  There's a bit of a memory increase per instance, but also there's quite a lot of processing that's done as a function of the max number of groups, so I'll want to carefully measure for any untoward performance hits.

I think the bigger problem is that groups aren't all that they might be, mainly due to a Reaper limitation, because Reaper itself doesn't understand Reaticulate groups, and Reaper will only ever chase the last program event on a given channel, which means chasing is broken when it comes to groups.  Having more groups will amplify this problem.

You can read more about this on issue #97, but as you can see there, trying to work around this limitation would require some heroics.

Still, I suppose with that accepted, increasing to 8 groups probably wouldn't be too much of a problem.   But I definitely can't support unlimited without significant code overhauls, and for a semi-working feature, it's definitely not worth doing that. :)

**Comment by sebastianruizva** [comment]
Thank you for considering this, I am sure 8 groups would suffice for most instruments. I love your work by the way, thanks for sharing it with the community!

**Comment by Vadline** [comment]
yes! 8 groups will be great for almost all cases (if it does not decrease performance)
